common/step_download.go

```diff
@@ -208,20 +192,19 @@ func (s *StepDownload) download(ctx context.Context, ui packer.Ui, source string
 	}
 
 	ui.Say(fmt.Sprintf("Trying %s", u.String()))
-	gc := getter.Client{
-		Ctx:              ctx,
+	req := &getter.Request{
 		Dst:              targetPath,
 		Src:              src,
 		ProgressListener: ui,
 		Pwd:              wd,
-		Dir:              false,
-		Getters:          getters,
+		Mode:             getter.ModeFile,
+		Inplace:          true,
 	}
 
-	switch err := gc.Get(); err.(type) {
+	switch op, err := defaultGetterClient.Get(ctx, req); err.(type) {
```

Comment:
Can't we use `getter.DefaultClient` here as well?

Reply:
In line with this question. On line 55 we create `defaultGetterClient` with no Detectors or Decompressors is this to fix another issue or disable default functionality that we don't want?

Reply:
@sylviamoss that's a very good question, I originally wanted to not use the DefaultClient because package scoped variables are sort of an anti pattern [link ](https://dave.cheney.net/2017/06/11/go-without-package-scoped-variables). I did implement it for the go-getter for convenience though and then I I recently used it in here:

https://github.com/hashicorp/packer/blob/cc3d941853c47fc099ba6bd3ea133d8e818cb67a/common/iso_config.go#L166

As I went a bit too fast over this !

@nywilken, these should be defaulted by the default func, which is often called to do init this.

https://github.com/hashicorp/packer/blob/a19214afeb6b294fe70dbeebe1c71192385efb1a/vendor/github.com/hashicorp/go-getter/v2/client_option.go#L3-L22

So, I'm going to not use DefaultClient in the other file instead :D

Reply:
Nice!! Thanks for explaining :D

Reply:
Thanks for noticing this :D